Oop-1 Question2 Matrix/Matrix.java

```diff
@@ -0,0 +1,184 @@
+/**
+ * This class is designed to work on matrices
+ * @author Gaurav Tak
+ *
+ */
+public class Matrix {
+	private int[][] matrixElement;
+	private int noOfRow;
+	private int noOfColumn;
+
+	public int getNoOfRow() {
+		return noOfRow;
+	}
+
+	public int getNoOfColumn() {
+		return noOfColumn;
+	}
+	
+	public Matrix() {
+		//do nothing
+	}
+	
+	public Matrix(int row, int column) {
+		noOfRow = row;
+		noOfColumn = column;
+		matrixElement = new int[noOfRow][noOfColumn];
+	}
+	
+	public Matrix(int[][] input) {
+		noOfRow = input.length;
+		noOfColumn = input[0].length;
+		matrixElement = new int[noOfRow][noOfColumn];
+		setMatrix(input);
+	}
+	
+	//getters and setters
+	
+	public void setMatrix(int[][] elements) {
+		if (noOfRow == elements.length || noOfColumn == elements[0].length)
+			for (int rowIndex = 0; rowIndex < noOfRow; rowIndex++) {
+				for (int columnIndex = 0; columnIndex < noOfColumn; columnIndex++) {
+					matrixElement[rowIndex][columnIndex] = elements[rowIndex][columnIndex];
+				}
+			}
+	}
+
+	public void setElement(int row, int column, int value) {
+		matrixElement[row][column] = value;
+	}
+	
+	public int getElement(int row,int col) {
+		return matrixElement[row][col];
+	}
+	
+	/**
+	 * to find transpose of a matrix
+	 * @return the tranposed 2d-array for the matrix
+	 * @throws NullPointerException
+	 */
+	public int[][] transpose() throws NullPointerException {
```

Comment:
this method is functionally correct but we can make it more simpler by using swap method